### PR TITLE
Modify Hospital beds Questionnaire 

### PR DESCRIPTION
**Pull request by yuliumitre**
Display the encounters within 6 months with encounter type and date. 

To test this rule changes, need to switch to the same branch for test-ehr and dtr.

### CODE REVIEW

File: HospitalBeds/R4/resources/Quesionnaire-R4-HospitalBeds.json [hunk 1]
```diff
@@ -142,7 +142,6 @@
     },
     {
       "linkId": "2",
-      "text": "Patient Diagnosis",
```
Comment (yunwwang) [inline HospitalBeds/R4/resources/Quesionnaire-R4-HospitalBeds.json:145]: Group should also have text for display.

File: HospitalBeds/R4/resources/Quesionnaire-R4-HospitalBeds.json [hunk 2]
```diff
@@ -161,20 +160,26 @@
...
+                  "valueExpression": {
+                    "language": "text/cql",
+                    "expression": "\"HospitalBedsPrepopulation\".CurrentDiagnoses"
+                  }
+                }
+              ],
+              "linkId": "2.2.1",
+              "text": "Diagnosis:",
+              "type": "open-choice",
+              "repeats": true,
```
Comment (yunwwang) [inline HospitalBeds/R4/resources/Quesionnaire-R4-HospitalBeds.json]: Don't need repeats:true for open-choice
Instead, using the extension
 { 
              "url": "http://hl7.org/fhir/StructureDefinition/questionnaire-answerRepeats", 
              "valueBoolean": true 
            }

File: HospitalBeds/R4/resources/Quesionnaire-R4-HospitalBeds.json [hunk 3]
```diff
@@ -188,13 +193,15 @@
...
+              "valueString": "\"HospitalBedsPrepopulation\".EncounterList"
             }
           ],
           "linkId": "3.1",
           "text": "Face-to-face Evaluation within 6 months",
-          "type": "boolean",
-          "required": false
+          "type": "open-choice",
+          "required": false,
+          "repeats": true,
```
Comment (yunwwang) [inline HospitalBeds/R4/resources/Quesionnaire-R4-HospitalBeds.json]: Don't need repeats:true for open-choice. Add extension mentioned in previous comment

File: HospitalBeds/R4/resources/Quesionnaire-R4-HospitalBeds.json [hunk 4]
```diff
@@ -220,44 +227,37 @@
...
+          }
+        },
+        {
+          "valueCoding": {
+            "code": "newSupplier",
+            "display": "New Supplier"
+          }
+        },
+        {
+          "valueCoding": {
```
Comment (yunwwang) [inline HospitalBeds/R4/resources/Quesionnaire-R4-HospitalBeds.json]: Add a separate text item for other

File: HospitalBeds/R4/resources/Quesionnaire-R4-HospitalBeds.json [hunk 5]
```diff
@@ -277,7 +277,7 @@
           "text": "Equipment",
           "type": "choice",
           "required": false,
-          "option": [
+          "answerOption": [
             {
               "valueCoding": {
                 "code": "E0250",
```
Comment (yunwwang) [inline HospitalBeds/R4/resources/Quesionnaire-R4-HospitalBeds.json:305]: There are bunch of different HCPCS codes in the template. Why are there only two in the questionnaire?

File: HospitalBeds/R4/resources/Quesionnaire-R4-HospitalBeds.json [hunk 6]
```diff
@@ -220,44 +227,37 @@
...
+          }
+        },
+        {
+          "valueCoding": {
+            "code": "other",
+            "display": "Other"
+          }
         }
       ]
     },
```
Comment (yunwwang) [inline HospitalBeds/R4/resources/Quesionnaire-R4-HospitalBeds.json:280]: Missing Order description between Item 5 and Item 6

File: HospitalBeds/R4/resources/Quesionnaire-R4-HospitalBeds.json [hunk 7]
```diff
@@ -293,13 +293,29 @@
...
       ]
-    },  
+    },
     {
       "linkId": "7",
       "type": "group",
       "item": [
         {
           "linkId": "7.1",
+          "text": "Other equipment, accessories, or supplies:",
```
Comment (yunwwang) [inline HospitalBeds/R4/resources/Quesionnaire-R4-HospitalBeds.json:319]: 7.1 is a description/title and should be combined with item 7

File: HospitalBeds/R4/resources/Quesionnaire-R4-HospitalBeds.json [hunk 8]
```diff
@@ -343,7 +359,7 @@
               }
             }
           ],
-          "linkId": "7.4",
+          "linkId": "8.4",
           "text": "NPI:",
           "type": "string",
           "required": true
```
Comment (yunwwang) [inline HospitalBeds/R4/resources/Quesionnaire-R4-HospitalBeds.json]: NPI is not required

File: HospitalBeds/R4/resources/Quesionnaire-R4-HospitalBeds.json [hunk 9]
```diff
@@ -293,13 +293,29 @@
...
+          "type": "string"
+        }
+      ]
+    },
+    {
+      "linkId": "8",
+      "type": "group",
+      "item": [
+        {
+          "linkId": "8.1",
```
Comment (yunwwang) [inline HospitalBeds/R4/resources/Quesionnaire-R4-HospitalBeds.json:334]: Signature is required

File: HospitalBeds/R4/resources/Quesionnaire-R4-HospitalBeds.json [hunk 10]
```diff
@@ -142,7 +142,6 @@
     },
```
Comment (yunwwang) [inline HospitalBeds/R4/resources/Quesionnaire-R4-HospitalBeds.json:141]: Item 1 should also have text.
1.4 and 1.5 are not required
1.5 should be answerOption

File: HospitalBeds/R4/resources/Quesionnaire-R4-HospitalBeds.json (hunk 5 above)
Comment (yuliumitre) [reply HospitalBeds/R4/resources/Quesionnaire-R4-HospitalBeds.json:305]: The story only requires to support E0250 and E0251 :-)

**Comment by yuliumitre** [comment]
@yunwwang I have addressed the comments. Let me know if I have missed anything. Thanks.

File: HospitalBeds/R4/resources/Quesionnaire-R4-HospitalBeds.json (hunk 3 above)
Comment (yunwwang) [reply HospitalBeds/R4/resources/Quesionnaire-R4-HospitalBeds.json]: Check the fix I created for previous one

**Comment by yunwwang** [comment]
There are two files for HomeOxygen Therapy. Are those part of this PR?

**Comment by yuliumitre** [comment]
I deleted the face to face questionnaire :-) Anything else? @yunwwang  Thanks a lot.

**Review by yunwwang — APPROVED** [review]

**Pull request merged**